tests/mocks.py

```diff
@@ -145,6 +147,19 @@ def revision(
             }
         )
 
+        next_free_txn_id = len(self._transactions)
+        for txn_id, comment in enumerate(comments, start=next_free_txn_id):
+            # See https://phabricator.services.mozilla.com/conduit/method/transaction.search/  # noqa
+            # for the Conduit API result object structure.
+            self._transactions.append(
```

Comment:
I wonder if it's worth sticking all the transaction generation logic into a `def transaction` on `PhabricatorDouble`? As we add more uses of transactions (there are a number of features coming that need it), we'll have to represent the other types. Would make this `PhabricatorDouble.revision` method simpler by having it just do something like `self.transaction(on=phid, comments=comments)`

Reply:
I agree, I've added the new method and the code for both adding transactions and creating revisions is now easier to read.